contracts/Registry.sol

```diff
@@ -1,70 +1,682 @@
+
+        for (uint i = 0; i < tags.length; i++) {
+            removeTagFromTypeRepositoryRegistration(orgName, repositoryName, tags[i]);
+        }
+
+        emit TypeRepositoryModified(orgName, repositoryName, orgName, repositoryName);
+    }
+
+    function removeTagFromTypeRepositoryRegistration(bytes32 orgName, bytes32 repositoryName, bytes32 tagName) internal {
+
+        // no-op if tag doesnt exist
+        if (orgsByName[orgName].typeReposByName[repositoryName].tagsByName[tagName].tagName != bytes32(0x0)) {
+
+            // swap type repository registration lut entries
+            uint tagIndexToReplace = orgsByName[orgName].typeReposByName[repositoryName].tagsByName[tagName].itemTagIndex;
+            bytes32 tagNameToMove = orgsByName[orgName].typeReposByName[repositoryName].tags[orgsByName[orgName].typeReposByName[repositoryName].tags.length-1];
+
+            // no-op if we are deleting the last item
+            if (tagIndexToReplace != orgsByName[orgName].typeReposByName[repositoryName].tags.length-1) {
+
+                orgsByName[orgName].typeReposByName[repositoryName].tags[tagIndexToReplace] = tagNameToMove;
+                orgsByName[orgName].typeReposByName[repositoryName].tagsByName[tagNameToMove].itemTagIndex = tagIndexToReplace;
+            }
+
+            orgsByName[orgName].typeReposByName[repositoryName].tags.length--;
+
+            // swap global tag index lut entries
+            tagIndexToReplace = orgsByName[orgName].typeReposByName[repositoryName].tagsByName[tagName].globalTagIndex;
+            uint tagIndexToMove = typeReposByTag[tagName].orgNames.length-1;
+
+            // no-op if we are deleting the last item
+            if (tagIndexToMove != tagIndexToReplace) {
+                bytes32 orgNameToMove  = typeReposByTag[tagName].orgNames[tagIndexToMove];
+                bytes32 repoNameToMove = typeReposByTag[tagName].itemNames[tagIndexToMove];
+
+                typeReposByTag[tagName].orgNames[tagIndexToReplace]  = orgNameToMove;
+                typeReposByTag[tagName].itemNames[tagIndexToReplace] = repoNameToMove;
+
+                orgsByName[orgNameToMove].typeReposByName[repoNameToMove].tagsByName[tagName].globalTagIndex = tagIndexToReplace;
+            }
+
+            typeReposByTag[tagName].orgNames.length--;
+            typeReposByTag[tagName].itemNames.length--;
+
+            // delete contents of the tag entry
+            delete orgsByName[orgName].typeReposByName[repositoryName].tagsByName[tagName];
+        }
+    }
+
+    function deleteTypeRepositoryRegistration(bytes32 orgName, bytes32 repositoryName) external {
+
+        requireOrgExistenceConstraint(orgName, true);
+        requireAuthorization(orgName, true);
+        requireTypeRepositoryExistenceConstraint(orgName, repositoryName, true);
+
+        deleteTypeRepositoryRegistrationInternal(orgName, repositoryName);
+
+        emit TypeRepositoryModified(orgName, repositoryName, orgName, repositoryName);
```

Comment:
`emit TypeRepositoryDeleted(...`

Reply:
yikes. my bad.

Reply:
tests also seem to take much longer with events enabled. will investigate.

Reply:
tests taking longer was unrelated to events.